vasp: change openacc cpp_options for vasp 6.5.0 or newer

saroj-lbl opened the pull request:
Implementing the cpp_options changed around 6.5.0 for openacc; see https://vasp.at/wiki/Makefile.include.nvhpc_omp_acc

These changes were needed for me to build 6.5.0 using nvhpc

saroj-lbl committed:
change openacc cpp_options for vasp 6.5.0 or newer

saroj-lbl commented:
@snehring

snehring commented:
I'm having a hell of a time getting this to compile +cuda in my env, but I think that's more a me problem than anything related to this. I'm assuming you've built and tested this?

saroj-lbl commented:
Yeah, there are additional changes needed to get the build done completely -- such as in the `setup_build_environment` I added env variables for `NVCOMPILER_COMM_LIBS_HOME`, `NVCOMPILER_NCCL_HOME`, `CUDA_HOME` etc.. and also removed the `depends_on("nccl")`. My goal was to use libraries from nvhpc as much as possible.

I could try to put all of that together in this PR or in a separate PR later.

snehring commented:
I suspect this is fine. It's certainly correct code wise, I've just not been able to dedicate time to make nvhpc work for me.

snehring reviewed (approved):
lgtm

snehring commented:
a separate PR to better handle the nvhpc integration when building +cuda would be appreciated.

saroj-lbl commented:
Yeah, sounds good. It is difficult to get the nvhpc integration in a generic way; currently, I have to hard-code the cuda version provided by the nvhpc version I am using. I am unsure about the timeline, but I will poke around at some point to see if I can do it in a more generic way.

saroj-lbl committed:
Merge branch 'develop' into vasp_cpp_options_update

saroj-lbl commented:
@snehring @alalazo let me know if anything is needed on my part before merging this request?
